### PR TITLE
Spawn loudspeaker at the roadblocks

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary
```SUMMARY: Content "Spawn loudspeaker at the roadblocks."```

#### Purpose of change
Address some issues concerning insufficient warnings leading to cheap death by turrets.

#### Describe the solution
- Added a `loudspeaker` "monster". For now it will spawn only at the roadblocks and will continually shout a set of warnings which should warn player about serious threat up ahead.
- Warnings are written as snippets, so they are easily could be added or edited.
- Warnings are chosen randomly from the list.
- Loudness for warning sound is 60, so it should be heard from the border of reality bubble.
- Cooldown for another warning is 10 seconds.
- Loudspeaker will spawn at the center of the roadblock.
- Loudspeaker will drop some low-importance loot on its death.

#### Describe alternatives you've considered
- Use `parrot` monster attack function instead of creating a new one. But this function for some reason make random-duration delays between warnings while I wanted fixed ones.
- Use existing `speaker cabinet` furniture instead of a new monster. But that would require a lot of new code to make furniture shout warnings.

#### Testing
Debug-spawned roadblocks, walked around a bit, listened for warnings.

#### Additional context
None.

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
> Would the search light bots have a role here too?

Well, searchlights are designed to find targets for turrets to destroy, while loudspeakers are designed to deter accidental survivors from entering dangerous territory.

**Comment by Zireael07** [comment]
Great, that should address the concerns we had in #35378 <3

File: src/monattack.cpp [hunk 1]
```diff
@@ -3522,6 +3522,13 @@ bool mattack::searchlight( monster *z )
     return true;
 }
 
+bool mattack::speaker( monster *z )
+{
+    sounds::sound( z->pos(), 60, sounds::sound_t::order, ( _( "%s" ),
+                   SNIPPET.random_from_category( "speaker_warning" ) ) );
```
Comment (BevapDin) [inline src/monattack.cpp]: This is not doing what you think it does.

This is a pointless translation. How is a translator supposed to know what they should translate? Why even translate this?
```C++
_( "%s" )
```

This invokes the comma-operator, which evaluates both arguments (calls `_` and `random_from_category`) and returns the value of the right operand and discards the value of the left operand. So the translations (if any) is discarded anyway.
```C++
( _( "%s" ), SNIPPET.random_from_category( "speaker_warning" ) )
```
```suggestion
    sounds::sound( z->pos(), 60, sounds::sound_t::order, SNIPPET.random_from_category( "speaker_warning" ) );
```

(Note that `random_from_category` returns an already translated string.)

**Comment by kevingranade** [comment]
I don't like the loudspeakers at all, the immediate effect is they're going to attract zombies to be shot and keep attracting them.  That would have been apparent the very first time they were deployed, so the question is, did the people who placed these roadblocks really want to attract endless hordes of zombies to their roadblocks? To me that seems incoherent.

How about having a light source in the roadblock area? This guarantees that in either daytime or nighttime, the player can see the turret first, it is a natural thing to have in a roadblock, and it won't trigger an unintended consequence of attracting hordes of zombies.

**Comment by I-am-Erk** [comment]
Adding a utility light and either an RTG generator hooked to it, with a small solar array, would have a couple effects:
- easier to see roadblocks
- adds more useful loot to players able to get past the significant defenses.

I agree that playing sound is not logical from a lore perspective.

**Comment by Night-Pryanik** [comment]
How I see it: the Cataclysm hasn't happened yet, but it is very close. Lots of yet living civilian people are panicking and run for their lives, mostly by the roads. Military established roadblocks on the roads to prevent these people from entering some forbidden areas. They deployed turrets and loudspeakers to deter the refugees.
Then the Cataclysm struck, and military who were guarding these roadblocks were either killed or deserted. There was no one left who cared for turning loudspeakers off and maybe deactivate turrets too.
In short, there were no zombie hordes yet at the time of deployment of loudspeakers and turrets. No one thought there would be lots of zombified humans in such short time.

File: data/json/snippets/snippets.json [hunk 2]
```diff
@@ -643,5 +643,16 @@
...
+  },
+  {
+    "type": "snippet",
+    "category": "speaker_warning",
+    "text": [
+      "WARNING CITIZENS!",
+      "THIS IS A RESTRICTED AREA!",
+      "NO ENTRY!",
+      "VIOLATORS WILL BE SHOT!",
+      "THIS IS A RECURRING MESSAGE!"
```
Comment (aoae23) [inline data/json/snippets/snippets.json]: Why this last snippet? It doesn't sound really fitting with the rest of them.

**Comment by I-am-Erk** [comment]
Setting up an automated loudspeaker at a road block still doesn't seem to make sense. Are you expecting the humans will all die? If so, you know there are zombies flooding the area and this will attract them. Are you expecting it will just stand behind your human workers and annoy and distract them? If so, why don't you have an officer with a megaphone?

Remember, the turrets aren't meant to be autonomous. The road block is meant to be manned by human staff. A loudspeaker is something easily done by a human.

From a gameplay perspective, this will have a terrible impact on balance, making roadblocks an incredibly easy way to deal with zombies.

**Comment by thethunderhawk** [comment]
An automated loudspeaker *would* make sense at a military staging area or refugee processing station, with automated messages giving instructions to arrivals. FEMA camps for example.

@I-Am-Erk I don’t exactly see how roadblocks which attract zombies using sound within the reality bubble would make dealing with zombies trivial. As of now the roadblocks already usually make a lot of noise when the turrets fire on random mobs. Baiting a group of zombies to a roadblock is possible but not trivial, since the zeds will prioritize chasing the player unless they can’t see them. 

It’s not hard to imagine a section of engineers setting up a bait-and-kill zone *assuming* autonomous turrets are available to them. Maybe there should be a rare roadblock variant with mines, turrets, and loudspeakers to represent that.

**Comment by I-am-Erk** [comment]
Automated loudspeakers would be immersive and creepy. I just don't see them being likely at roadstops, where most of the time they'd be an unwelcome distraction to the people actually operating the roadstop, who would eventually realize they were also a deadly liability. I could see them being scattered throughout cities, looping emergency broadcasts for a few days after the cataclysm until their batteries run down, or looping emergency messages by day and for a few hours at night until they could recharge by solar panel. Then we could also have a map extra inside cities where survivors/cops gathered a few emergency loudspeakers into a bundle surrounded by turrets and barbed wire, as a zombie trap.

**Comment by kevingranade** [comment]
> An automated loudspeaker would make sense at a military staging area or refugee processing station, with automated messages giving instructions to arrivals. FEMA camps for example.

Absolutely, there are places it would make sense to place loudspeakers IF you don't know it will attract zombies OR you WANT them to attract zombies, but they make no sense at a roadblock.

For example, stick some loudspeakers at the FEMA camp.

> It’s not hard to imagine a section of military engineers quickly setting up a bait-and-kill zone outside a town (assuming autonomous turrets are available to them). Maybe there should be a rare roadblock variant with mines, turrets, and loudspeakers to represent that.

That's a completely different thing that we're not talking about, if you want to add something that is a "zombie kill station" that has an audible signal for attracting zombies and repelling humans, that's fine, but it's not something you would do at the existing locations, plus if you're actively trying to drive zombies to it, it would be exhausted of ammo and overrun by the time the game starts.

**Comment by I-am-Erk** [comment]
Actually I totally want to add a zombie kill station and am going to do that right now.

**Comment by davidpwbrown** [comment]
fwiw I thought it coulda worked, with the justification Night-Priyanik gave.
